capa/helpers.py

```diff
@@ -126,33 +126,46 @@ def new_print(*args, **kwargs):
         inspect.builtins.print = old_print  # type: ignore
 
 
-def log_unsupported_format_error():
-    logger.error("-" * 80)
-    logger.error(" Input file does not appear to be a PE or ELF file.")
-    logger.error(" ")
-    logger.error(
-        " capa currently only supports analyzing PE and ELF files (or shellcode, when using --format sc32|sc64)."
-    )
-    logger.error(" If you don't know the input file type, you can try using the `file` utility to guess it.")
-    logger.error("-" * 80)
+def exceptUnsupportedError(func):
```

Comment:
let's use snake case names and maybe rename this to `catch_log_return_errors` or similar?

then let's use via `@<decorator_name>` (see, e.g., https://rinaarts.com/declutter-python-code-with-error-handling-decorators/)